### ISSUE TITLE
Exported gzip file of database has first ~73 kB uncompressed and rest is gzip compressed

### ISSUE THREAD

**Issue by prcdpr** [phpmyadmin/phpmyadmin#16042]
### Describe the bug

I'm exporting 280 MB database with Export to file with gzip compression.
Resulting file `database.sql.gz` has first 73 kB written in plain SQL and then the rest of file is gzip compressed.

### To Reproduce

1. Export database to file with gzip compression checked.
2. Leave all other export settings default

### Expected behavior

Resulting export file should be fully gzip compresed.

### Server configuration

- Operating system: Debian 8
- Web server: nginx/1.6.2 
- Database version: 8.0.13 - MySQL Community Server - GPL 
- PHP version: 7.4.4 
- phpMyAdmin version: 4.9.5

### Client configuration

- Browser: Firefox
- Operating system: Win10

**Comment by williamdes** [comment]
Thank you, I will try to reproduce this issue

**Comment by ibennetch** [comment]
Thanks for your good report, this is strange as I can't reproduce this. I'm trying with the sakila test database using the payment table, which phpMyAdmin reports as approximately 2.1 MiB, and when compressed gives me a 235Kb file.

Can you try on the demo server at https://demo.phpmyadmin.net/QA_4_9/ using username root with a blank password?

**Comment by prcdpr** [comment]
I have troubles reproducing it on this demo server so I'll try again tomorrow.

For now I know that compressed part starts always at the same offset no matter which tables I select for export. To decompress compressed part it's enough to call
```
tail -c +75370 ./database.sql.gz | gunzip
```

So first 75370 bytes are uncompressed, then the rest of file is compressed.

**Comment by prcdpr** [comment]
On Chrome it's fully compressed! So it's a Firefox problem.

**Comment by ibennetch** [comment]
Thanks for the additional information. That's so wacky! I'm using
Firefox 74 here on Mac, I'll have to test some other combinations.

**Comment by prcdpr** [comment]
Might be related:

https://bugzilla.mozilla.org/show_bug.cgi?id=424306
https://bugzilla.mozilla.org/show_bug.cgi?id=233047

Workaround: in `phpMyAdmin-4.9.5-all-languages\libraries\classes\Export.php` change

```
$mime_type = 'application/x-gzip';
```
to 

```
$mime_type = 'application/octet-stream';
```

And it works!

But 2004 Firefox bug, really...?

**Comment by Faber44** [comment]
Got the same problem with Firefox Developper 96 + phpMyAdmin 5.1.1
It worked for me. Thanks! (y)

**Comment by OldVamp** [comment]
had same issue with firefox 95.0.1, editing export.php fixed it

**Comment by simonchilton** [comment]
I don't know if this is related but I'm getting a similar problem in Firefox 99.0.1 on MacOS 12.3.1 and phpMyAdmin 5.1.3.

When I export tables within a database with tar.gz compression selected, Firefox only downloads the first 72.9kb (the export should be about 3MB) and the resultant file is not a complete archive.

**Comment by viktorhaid** [comment]
We are facing a similar problem using Chrome (Version Version 101.0.4951.67), Windows 10, phpmyadmin 5.1.3. 
The export file has a gz extension but the file content is not compressed and cut down to ~73 kb.

**Comment by fox91** [comment]
I've tested some versions of phpMyAdmin and common browsers and I can confirm the bug:

- `4.6.6`, `4.7.9`, `4.8.5`, `4.9.7`, `5.0.4`
    - :white_check_mark: Chromium `106.0.5249.91`
    - :white_check_mark: Firefox `78.15.0esr`
    - :x: Firefox `94.0`: invalid gzip, file truncated at ~75KB
    - :x: Firefox `105.0`: invalid gzip, file truncated at ~75KB
- `5.1.0`, `5.1.1`
    - :white_check_mark: Chromium `106.0.5249.91`
    - :white_check_mark: Edge `106.0.1370.37`
    - :white_check_mark: Firefox `78.15.0esr`
    - :x: Firefox `94.0`: invalid gzip, file truncated at ~64MB
    - :x: Firefox `105.0`: invalid gzip, file truncated at ~64MB
- `5.1.2`, `5.1.3`, `5.1.4`
    - :x: Chromium `106.0.5249.91`: invalid gzip, file truncated at ~64MB)
    - :x: Edge `106.0.1370.37`: invalid gzip, file truncated at ~64MB
    - :white_check_mark: Firefox `78.15.0esr`
    - :x: Firefox `94.0`: invalid gzip, file truncated at ~64MB
    - :x: Firefox `105.0`: invalid gzip, file truncated at ~64MB
- `5.2.0`
    - :white_check_mark: Chromium `106.0.5249.91`
    - :white_check_mark: Edge `106.0.1370.37`
    - :white_check_mark: Firefox `78.15.0esr`
    - :x: Firefox `94.0`: invalid gzip, file truncated at ~64MB
    - :x: Firefox `105.0`: invalid gzip, file truncated at ~64MB

I've tested both Ubuntu and Windows browser versions.

phpMyAdmin started from official docker images (or old community when not available).

**Comment by Emil-Av** [comment]
I have the same problem with FireFox 106.0.2 and phpMyAdmin 5.2. Exporting a database from the hosting provider Strato results in invalid gzip, file truncated at  ~73KB. Exporting from the local phpMyAdmin with the same versions works though.. 

Exporting from Strato with Chrome works fine as well.

**Comment by williamdes** [comment]
So if I do testing I should have the same file fingerprint between browsers, right?

I assigned this to the milestone so we have a look in priority at what can be done
@kamil-tekiela would you want to contribute to this one?

**Comment by kamil-tekiela** [comment]
This is the third time I have been looking at the same piece of code. This time I decided to dig up the history to understand why it's there. Here are some links:

- https://bugs.chromium.org/p/chromium/issues/detail?id=268085
- https://github.com/phpmyadmin/phpmyadmin/commit/ccd65816b8420b5e5239537bb110185d7ac2cace (Initial fix. I don't think this fix was the correct one. It seems to incorrectly set the header to trick buggy Apache 2.2 servers)
- https://github.com/phpmyadmin/phpmyadmin/commit/a39a5ce2656a8a6648985bfbe6ed4aa65ba3b27c (a fix for a fix. Chrome patched the bug up and phpMyAdmin stopped this hack from happening in Chrome. Now it seems Firefox has also caught up and fixed the bug)
- https://github.com/phpmyadmin/phpmyadmin/issues/11283

Recently, we doubled-down on this fix because PHP changed the behaviour of output buffering compression. So to fix this properly, I suggest to remove the initial hack and the follow up hack with it, but leave the hack for PHP output buffering... if that makes sense.

**Comment by williamdes** [comment]
Thank you @kamil-tekiela for your great work and investigation !
The the latest 5.2 version in development [(phpMyAdmin 5.2+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.2+snapshot) is updated, you can test it and let us know here

**Comment by TheOldMan2000** [comment]
Hi,

Is there a patch/fix for the PhpMyAdmin 4.9.7 release available, please?

Many thanks.



Issue: 

I'm experiencing this in Firefox 108.0.2 (64-bit) and Google Chrome Version 108.0.5359.125, 
(Official Build) (64-bit) on Windows 10 Home 22H2 (OS Build 19045.2364, 
Windows Feature Experience Pack 120.2212.4190.0).

I downloaded a 191.2 MB database export:

> **Firefox 108**
> .gz version size **is only 73.1 KB (74,940 bytes)**
> .zip version size is 89.3 MB (93,658,648 bytes)

> **Chrome Version 108**
> .gz version size is 90.7 MB (95,181,552 bytes)
> .zip version size is 89.2 MB (93,582,344 bytes)

The gzipped actual file in FF takes its time to download as normal would expect, but as soon as its done the download is corrupt at 74KB. Apache server has gzip enabled working when I browse sites etc.

If it helps at all:

> Latest WHM/Cpanel 106 Build 11 uses PhpMyAdmin 4.9.7:
> cpsrvd 11.106.0.11
> Database client version: libmysql - mysqlnd 7.4.33
> PHP version: 7.4.33
> 
>     Server: Localhost via UNIX socket
>     Server type: MariaDB
>     Server connection: SSL is not being used Documentation
>     Server version: 10.6.11-MariaDB-log - MariaDB Server
>     Protocol version: 10
>     User: cpses_xxxxxx7xxxx@localhost
>     Server charset: UTF-8 Unicode (utf8mb4)
> 
> 
> Server:
> System: Linux vpsxxxxx.xxx.com 3.10.0-1160.53.1.vz7.185.3 #1 SMP Tue Jan 25 12:49:12 MSK 2022 x86_64
> Build Date: Jan 4 2023 02:32:39
> Build System: Linux hou-2 5.3.18-150300.59.98-default #1 SMP Thu Oct 13 08:52:00 UTC 2022 (dfcde7e) x86_64 x86_64 x86_64 GNU/Linux
> PHP 8.1.13 FPM/FastCGI 
> 
> Brotli support | enabled
> Extension Version | 0.13.1
> Library Version | 1.0.9
> 
> BZ2
> BZip2 Support | Enabled
> Stream Wrapper support | compress.bzip2://
> Stream Filter support | bzip2.decompress, bzip2.compress
> BZip2 Version | 1.0.6, 6-Sept-2010
> 
> ZIP
> Zip | enabled
> Zip version | 1.19.5
> Libzip version | 1.9.2
> BZIP2 compression | Yes
> XZ compression | Yes
> ZSTD compression | No
> AES-128 encryption | Yes
> AES-192 encryption | Yes
> AES-256 encryption | Yes
> 
> ZLIB
> Stream Wrapper | compress.zlib://
> Stream Filter | zlib.inflate, zlib.deflate
> Compiled Version | 1.2.7
> Linked Version | 1.2.7

**Comment by TheOldMan2000** [comment]
Apologies I see its fixed in 5.2.1. Hopefully Cpanel will catch up!

**Comment by williamdes** [comment]
> Is there a patch/fix for the PhpMyAdmin 4.9.7 release available, please?

I think you can apply https://github.com/phpmyadmin/phpmyadmin/commit/dca4119c1f2eac736c35eb2d4a5e4073d3f29790#diff-7904aae19b48949239d69719e6f1c29670b3de55bde4caaf59bfcddfd9a63994

**Comment by TheOldMan2000** [comment]
Great, thank you!
